gltf-importer/src/lib.rs

```diff
@@ -66,143 +68,52 @@ pub enum Error {
+    let data = read_to_end(path)?;
+    if data.starts_with(b"glTF") {
+        import_binary(&data, &config, path)
+    } else {
+        import_standard(&data, &config, path)
     }
+}   
 
-    /// Splits loaded GLB into its three chunks.
-    ///
-    /// * Mandatory GLB header.
-    /// * Mandatory JSON chunk.
-    /// * Optional BIN chunk.
-    fn split(&self) -> Result<Split, Error> {
-        use std::mem::{size_of, transmute};
-        let err = |reason: &str| Err(Error::MalformedGlb(reason.to_string()));
-        let glb = &self.0;
-        let glb_ptr = glb.as_ptr();
-        if glb.len() < size_of::<Header>() + size_of::<ChunkInfo>() {
-            return err("GLB missing header");
-        }
-
-        // Offset in bytes into `glb`.
-        let mut offset = 0isize;
-
-        let header = unsafe {
-            let x: *const Header = transmute(glb_ptr.offset(offset));
-            if &(*x).magic != b"glTF" {
-                return err("GLB does not start with `glTF`");
-            }
-            if (*x).length as usize != glb.len() {
-                return err("length does not match length of data");
-            }
-            if (*x).version != 2 {
-                return err("Not GLB version 2.0");
-            }
-            *x
-        };
-
-        offset += size_of::<Header>() as isize;
-        let json_chunk = unsafe {
-            let x: *const ChunkInfo = transmute(glb_ptr.offset(offset));
-            offset += size_of::<ChunkInfo>() as isize;
-            if (*x).length as usize > (glb.len() as isize - offset) as usize {
-                return err("JSON chunkLength exceeds length of data");
-            }
-            if &(*x).kind != b"JSON" {
-                return err("JSON chunkType is not `JSON`");
-            }
-            Slice {
-                offset: offset as usize,
-                length: (*x).length as usize,
-            }
-        };
-
-        offset += json_chunk.length as isize;
-        let blob_chunk = if glb.len() as isize - offset == 0 {
-            None
-        } else {
-            unsafe {
-                let x: *const ChunkInfo = transmute(glb_ptr.offset(offset));
-                offset += size_of::<ChunkInfo>() as isize;
-                if (*x).length as usize > (glb.len() as isize - offset) as usize {
-                    return err("BIN chunkLength exceeds length of data");
-                }
-                if &(*x).kind != b"BIN\0" {
-                    return err("BIN chunkType is not `BIN\0`");
-                }
-                Some(Slice {
-                    offset: offset as usize,
-                    length: (*x).length as usize,
-                })
-            }
-        };
-
-        Ok((header, json_chunk, blob_chunk))
-    }
+/// Imports glTF 2.0
+pub fn import<P>(path: P) -> Result<(Gltf, Buffers), Error>
+    where P: AsRef<Path>
+{
+    import_impl(path.as_ref(), Default::default())
```

Comment:
It seemed odd to have a `Config` struct with only one parameter, so I left it out to think about it some more. There's been no update since though so I guess we could just add an `import_with_config()` function?